kernel/src/process.rs

```diff
@@ -769,6 +829,11 @@ impl<C: Chip> ProcessType for Process<'a, C> {
     }
 
     fn setup_mpu(&self) {
+        // Do not modify an inactive process.
```

Comment:
I'm curious about the proliferation of these:
 - Are all the functions with these checks actually getting called on stopped processes?
 - Should that be a bug rather than a silent ignore?
 - How do we make sure we don't miss these in future process management stuff?

Reply:
>     * Are all the functions with these checks actually getting called on stopped processes?

Well the one that initiated this change is, notably if a capsule holds a copy of an appid for a stopped process and then uses it to enter its grant region. The others are not as most are called as a result of the process being scheduled, which it won't since it is stopped.

>     * Should that be a bug rather than a silent ignore?

In the case of entering a grant region it returns an error to the capsule.

For the others, maybe? It adds complexity in the common case. Maybe the better approach (at least for now) is to make the documentation clear that calling the functions without return codes is only valid when the process is active.



>     * How do we make sure we don't miss these in future process management stuff?

I'm not sure. Generally, functions with return values need to handle this, as those return values likely are passed (eventually) outside of the kernel crate. Functions without return values tend to be only called inside the kernel.

This is also complicated a little by code like the process console which does obtain references to `ProcessType` objects. But that is really a separate issue.